GymBuddies/ViewCells/RoutineTableViewCell.m

```diff
@@ -56,6 +58,54 @@ - (void)setCellContent:(Routine *)routine{
     self.workoutPlaceLabel.text = [SegmentedControlBlocksValues setWorkoutPlaceLabelContent:self.routine.workoutPlace];
     self.workoutPlaceLabel.layer.masksToBounds = YES;
     
+    [self setLikedRoutine];
+}
+
+
+-(void)setLikedRoutine{
+    [ParseAPIManager isLiked:self.routine completion:^(PFObject * _Nonnull object, NSError * _Nullable error) {
+        if(error == nil){
+            [self.likeButton setImage:[UIImage systemImageNamed:kLikedFilledRoutineButtonImage] forState:UIControlStateNormal];
+            self.likeButton.tintColor = [UIColor systemRedColor];
+            self.isLiked = YES;
+        } else{
+            [self.likeButton setImage:[UIImage systemImageNamed:kLikedNormalRoutineButtonImage] forState:UIControlStateNormal];
+            self.likeButton.tintColor = [UIColor systemBlueColor];
+            self.isLiked = NO;
+        }
+    }];
+}
+
+
+#pragma mark - Like method
+
+- (IBAction)didTapLike:(id)sender {
+    [self likeAction];
+}
+
+
+-(void)likeAction{
+    if(self.isLiked){
+        [self.likeButton setImage:[UIImage systemImageNamed:kLikedNormalRoutineButtonImage] forState:UIControlStateNormal];
```

Comment:
overall, logic between `if` and `else` blocks are completely the same, you should provide a helper method instead that will take the minimum number of changed parameters (like tintColor, likeCount, isLiked) based on `self.isLiked` or not.